microsoft/botframework-solutions
# [Enterprise Bot Template][TS] Add TypeScript as dependency

## Description

- Add TypeScript as dev dependency in the enterprise bot solution in order to avoid requiring TypeScript being installed globally. Additionally, this let specify the version of TypeScript that is being used.
- Add gitignore file to ignore generated files.

## Related Issue
Fixes https://github.com/Microsoft/AI/issues/546 [Enterprise Bot Template][TS] Avoid requiring global dependencies for building

## Testing Steps



1. Open a new terminal in AI/templates/Enterprise-Template/src/typescript.
1. Run `npm install` to install dependencies.
1. Run `npm run build` to build the solution
1. Check there is no exceptions encountered  even when you don't have typescript installed globally (i.e. using `npm install -g typescript`).

## Checklist
~~- [ ] I have commented my code, particularly in hard-to-understand areas~~
~~- [ ] I have added or updated the appropriate unit tests~~
~~- [ ] I have updated related documentation~~

If this contains changes that needs to be replicated between the Enterprise Template <-> Virtual Assistant
~~- [ ] A duplicate issue is filed to track future  work.~~

If you have updated responses or `.lu` files:
~~- [ ] All languages have been updated~~
~~- [ ] You have tested deployment with your new models~~
